client/src/ast.js

```diff
@@ -186,7 +187,14 @@ export function applyChange(arr, change) {
     } else {
       // If we don't have an offset, find the old val and
       // replace it with the new val
-      const index = arr.findIndex(x => x.id === change.old_val.id)
+      const index = arr.findIndex(x => deepEqual(x.id, change.old_val.id))
```

Comment:
@deontologician don't you need the same fix up in the `remove/uninitial` case on L161?